source/components/tooltip/HxTooltip.js

```diff
@@ -0,0 +1,191 @@
+import { getPosition } from '../../lib/position';
+import debounce from 'lodash/debounce';
+
+window.addEventListener('WebComponentsReady', function () {
+    const tagName = 'hx-tooltip';
+    const template = document.createElement('template');
+
+    template.innerHTML = `
+      <style>${require('./HxTooltip.less')}</style>
+      ${require('./HxTooltip.html')}`;
+
+    class HxTooltip extends HTMLElement {
+        static get is () {
+            return tagName;
+        }
+
+        static get observedAttributes () {
+            return [ 'open' ];
+        }
+
+        constructor () {
+            super();
+            this.attachShadow({ mode: 'open' });
+            if (window.ShadyCSS) {
+                ShadyCSS.prepareTemplate(template, tagName);
+                ShadyCSS.styleElement(this);
+            }
+            this.shadowRoot.appendChild(template.content.cloneNode(true));
+            this.initialPosition = this.position;
+            this._show = this._show.bind(this);
+            this._hide = this._hide.bind(this);
+            this._toggle = this._toggle.bind(this);
+            this._reposition = this._reposition.bind(this);
+            this._closeOnBackgroundClick = this._closeOnBackgroundClick.bind(this);
+        }
+
+        connectedCallback () {
+            this._target = document.querySelector('[data-tooltip=' + this.id + ']');
```

Comment:
Please document that you shouldn't use multiple targets with the same `data-tooltip` attribute value.